core/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -188,6 +192,44 @@ class DeltaCatalog extends DelegatingCatalogExtension
     }
   }
 
+  override def loadTable(ident: Identifier, timestamp: Long): Table = {
+    loadTableWithTimetravel(ident, version = None, Some(timestamp))
+  }
+
+  override def loadTable(ident: Identifier, version: String): Table = {
+    loadTableWithTimetravel(ident, Some(version), timestamp = None)
+  }
+
+  /**
+   * Helper method which loads a Delta table with given time travel parameters.
+   * Exactly one of the timetravel parameters (version or timestamp) must be present.
+   *
+   * @param version The table version to load
+   * @param timestamp The timestamp for the table to load, in microseconds
+   */
+  private def loadTableWithTimetravel(
+      ident: Identifier,
+      version: Option[String],
+      timestamp: Option[Long]): Table = {
+    val table = loadTable(ident)
+    assert(version.isEmpty ^ timestamp.isEmpty,
+      "Either the version or timestamp should be provided for time travel")
+    table match {
+      case deltaTable: DeltaTableV2 =>
+        val ttOpts = Map(DeltaDataSource.TIME_TRAVEL_SOURCE_KEY -> "SQL") ++
+          (if (version.isDefined) {
+            Map(DeltaDataSource.TIME_TRAVEL_VERSION_KEY -> version.get)
+          } else {
+            val timestampMs = timestamp.get / 1000
+            Map(DeltaDataSource.TIME_TRAVEL_TIMESTAMP_KEY -> new Timestamp(timestampMs).toString)
+          })
+
+        deltaTable.withOptions(ttOpts)
+      case default =>
+        throw new UnsupportedOperationException(s"time travel is not supported on: $default.")
```

Comment:
Should we just return None here and let `loadTable` call `super.loadTable`?

Reply:
@zsxwing will other tables that aren't DeltaTableV2 support time travel? how would I best test this?

Reply:
> will other tables that aren't DeltaTableV2 support time travel? how would I best test this?

This is just to let Spark decide what to do for non Delta tables. Maybe try to use `parquet` to see if it will hit this code path? If not, probably not worth to test that.